### PR TITLE
Move base test classes into the _cts module

### PR DESCRIPTION
**Pull request by clumens**
I could probably keep working on this one forever.  There's likely tons of stuff I've missed that I might like to do, but you've gotta stop at some point.  This just handles the base classes.  The tests themselves will come in a later PR.

I've tested this decently, but not exhaustively.

### CODE REVIEW

File: python/pacemaker/_cts/tests/base.py [hunk 1]
```diff
@@ -0,0 +1,915 @@
+""" Base classes for CTS tests """
+
+__all__ = ["CTSTest", "RemoteDriver", "SimulStartLite", "SimulStopLite", "StartTest", "StopTest"]
```
Comment (kgaillot) [inline python/pacemaker/_cts/tests/base.py:3]: Since we're doing a tests submodule, I'm thinking one source file per test class. It'll be a lot of files but very easy to tell what you're looking for. You can do that in separate commits so as not to have to separate out all the work already done.

File: python/pacemaker/_cts/tests/base.py [hunk 2]
```diff
@@ -0,0 +1,915 @@
...
+#        SPECIAL NOTE:
+#
+#        Tests may NOT implement any cluster-manager-specific code in them.
+#        EXTEND the ClusterManager object to provide the base capabilities
+#        the test needs if you need to do something that the current CM classes
+#        do not.  Otherwise you screw up the whole point of the object structure
+#        in CTS.
+#
+#                Thank you.
+#
```
Comment (kgaillot) [inline python/pacemaker/_cts/tests/base.py:17]: I'm thinking we can just drop this note. Hopefully once everything's better organized it will be more obvious where to do what. Alternatively we could add a section about CTS to Pacemaker Development and include it there.

File: cts/lab/CTStests.py [hunk 3]
```diff
@@ -161,9 +162,9 @@ def __call__(self, node):
             # 255 == broken pipe, ie. the node was fenced as expected
             self._logger.log("Locally originated fencing returned %d" % rc)
 
-        self.set_timer("fence")
-        matched = watch.look_for_all()
-        self.log_timer("fence")
+        with Timer(self._logger, self.name, "test"):
```
Comment (kgaillot) [inline cts/lab/CTStests.py:165]: Was this supposed to be "fence"?

**Review by kgaillot — COMMENTED** [review]
Whew! Looks great

File: cts/lab/CTStests.py (hunk 3 above)
Comment (clumens) [reply cts/lab/CTStests.py:165]: Yep, fixed.

File: python/pacemaker/_cts/tests/base.py (hunk 2 above)
Comment (clumens) [reply python/pacemaker/_cts/tests/base.py:17]: Fixed with an additional commit.

File: python/pacemaker/_cts/tests/base.py (hunk 1 above)
Comment (clumens) [reply python/pacemaker/_cts/tests/base.py:3]: Added a new commit to split everything up.  I wanted to do it in separate commits for each, but there's really nothing interesting to it and I didn't want to add a bunch of intermediate importing code that I'd just get rid of in a later patch.

Comment (clumens) [reply python/pacemaker/_cts/tests/base.py:3]: I've left CTSTest in base.py, but I'm not thrilled.  I could rename that file to ctstest.py, or put it in `__init__.py` (which I am also not thrilled about).

**Comment by clumens** [comment]
The other thing that bugs me here is the tests directory vs. the test.py file.  But those are good names for both.

File: python/pacemaker/_cts/tests/base.py (hunk 1 above)
Comment (kgaillot) [reply python/pacemaker/_cts/tests/base.py:3]: I though the convention was naming single-class files after the class (like CTSTest.py), but I guess there's no official recommendation

Comment (clumens) [reply python/pacemaker/_cts/tests/base.py]: Also moved what's left in base.py to ctstest.py.

File: cts/lab/CTStests.py [hunk 4]
```diff
@@ -26,7 +26,7 @@
 from pacemaker import BuildOptions
 from pacemaker._cts.CTS import NodeStatus
 from pacemaker._cts.audits import AuditResource
-from pacemaker._cts.tests.base import CTSTest, RemoteDriver, SimulStartLite, SimulStopLite, StartTest, StopTest
+from pacemaker._cts.tests import CTSTest, RemoteDriver, SimulStartLite, SimulStopLite, StartTest, StopTest
```
Comment (kgaillot) [inline cts/lab/CTStests.py:29]: I'm usually not a fan of `import *` but I think you're going to need that here when we add a zillion more tests. Not necessary for this PR though

**Review by kgaillot — APPROVED** [review]
